src/main/java/jenkinsci/plugins/influxdb/generators/JUnitPointGenerator.java

```diff
@@ -0,0 +1,85 @@
+package jenkinsci.plugins.influxdb.generators;
+
+import hudson.EnvVars;
+import hudson.model.BooleanParameterValue;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.tasks.junit.CaseResult;
+import hudson.tasks.test.AbstractTestResultAction;
+import jenkinsci.plugins.influxdb.renderer.MeasurementRenderer;
+import org.influxdb.dto.Point;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class JUnitPointGenerator extends AbstractPointGenerator{
+
+    private static final String JUNIT_SUITE_NAME = "suite_name";
+    private static final String JUNIT_TEST_NAME = "test_name";
+    private static final String JUNIT_TEST_STATUS = "test_status";
+
+    private final String customPrefix;
+    private final TaskListener listener;
+
+    private final EnvVars env;
+
+    public JUnitPointGenerator(Run<?, ?> build, TaskListener listener,
+                               MeasurementRenderer<Run<?, ?>> projectNameRenderer,
+                               long timestamp, String jenkinsEnvParameterTag,
+                               String customPrefix, EnvVars env) {
+        super(build, listener, projectNameRenderer, timestamp, jenkinsEnvParameterTag);
+        this.customPrefix = customPrefix;
+        this.listener = listener;
+        this.env = env;
+    }
+
+    /**
+     * @return true, if environment variable LOG_JUNIT_RESULTS is set to true and JUnit Reports exist
+     */
+    @Override
+    public boolean hasReport() {
+        return Boolean.parseBoolean(env.getOrDefault("LOG_JUNIT_RESULTS", "false")) && hasTestResults(build);
+    }
+
+    @Override
+    public Point[] generate() {
+
+        List<Point> points = new ArrayList<>();
+
+        // iterate each caseResult to get suiteName, testName and testStatus
+        List<CaseResult> allTestResults = getAllTestResults(build);
+
+        for (CaseResult caseResult : allTestResults) {
+            Point point = buildPoint("junit_data", customPrefix, build)
+                    .addField(JUNIT_TEST_NAME, caseResult.getSuiteResult().getName())
+                    .tag(JUNIT_SUITE_NAME, caseResult.getSuiteResult().getName())
+                    .addField(JUNIT_TEST_NAME, caseResult.getDisplayName())
+                    .tag(JUNIT_TEST_NAME, caseResult.getDisplayName())
+                    .addField(JUNIT_TEST_STATUS, caseResult.getStatus().toString())
+                    .tag(JUNIT_TEST_STATUS, caseResult.getStatus().toString())
+                    .addField("test_status_ordinal", caseResult.getStatus().ordinal())
+                    .build();
+            points.add(point);
+        }
```

Comment:
Does this work with multiple tests? InfluxDB uses time as a primary key and now you're creating multiple points without changing the timestamp. `AbstractPointGenerator` stores the wanted `timestamp`, so before the loop you could add something like `TimeGenerator testTimestamp = new TimeGenerator(timestamp);` and call the 4 parameter version of `buildPoint()` from `AbstractPointGenerator` to use a custom timestamp for your measurements.

Reply:
It did work with the 10 tests I had in my test project, but I understand your concern. 
According to the [InfluxDB documentation](https://docs.influxdata.com/influxdb/v1.8/troubleshooting/frequently-asked-questions/#how-does-influxdb-handle-duplicate-points) a point is uniquely identified by measurement name, tag set and timestamp. Measurement name and timestamp is the same for all datapoints here, but the tag set differs as each test has a unique test name.

Reply:
Ah, good to know. That's something I've overlooked. Most of this plugin's measurements have limited tags, so enforcing a different timestamp has been mandatory.